Scripts/Runtime/Data/VirtualData/VirtualData.cs

```diff
@@ -143,35 +158,156 @@ internal sealed override JobHandle ConsolidateForFrame(JobHandle dependsOn)
         //*************************************************************************************************************
 
         [BurstCompile]
-        private struct ConsolidateLookupJob : IJob
+        private struct ConsolidateLookupJob : IAnvilJob
         {
             private UnsafeTypedStream<TInstance> m_Pending;
-            private DeferredNativeArray<TInstance> m_Iteration;
-            private UnsafeParallelHashMap<TKey, TInstance> m_Lookup;
+            private DeferredNativeArray<TInstance> m_IterationTarget;
+            private UnsafeParallelHashMap<VDContextID, TInstance> m_Lookup;
+            private VDLookupReader<bool> m_CancelledLookup;
+            //TODO: DISCUSS - Split this out into separate data 
```

Comment:
This one requires a bit of forward thinking to actual use cases rather than my contrived Wander example. 

In the Wander, we have two competing flows. One where the Actor waits a bit, chooses a spot on the map, pathfinds to that spot and then walks to it and when they get there, repeats and starts waiting again.

The other flow is just a timer where once it expires, it cancels that Actor's Wander. Which means that the following structs get wiped (if they exist) for that Actor's Entity.

- WanderTaskInstance
- TimerInstance/Result
- PathfindRequest/Result
- FollowpathInstance/Result

That's all fine and expected. The WanderTaskDriver has a job to handle when it gets cancelled where it wants to restart itself. 

```
ConfigureCancelJob(HandleCancelJob.Schedule)
               .ScheduleOnForCancel(System.WanderInstanceData, BatchStrategy.MaximizeChunk)
               .RequireCancelledDataForIterateAsync(System.WanderInstanceData)
               .RequireDataForAddAsync(WanderStartData)
               .RequireDataForAddAsync(System.WanderInstanceData);
```

Here's we're using the cancelled instances of `WanderInstanceData` as the trigger for the job and what we want to iterate through. 

However since we want to restart, we need to create a new instance of `WanderInstanceData` and now we have a conflict in trying to access the same data in two different contexts in the same job. 

To get around it we can do an intermediate job to trigger the restart like so:

```
ConfigureCancelJob(HandleCancelJob.Schedule)
               .ScheduleOnForCancel(System.WanderInstanceData, BatchStrategy.MaximizeChunk)
               .RequireCancelledDataForIterateAsync(System.WanderInstanceData)
               .RequireDataForAddAsync(RestartData);
```

With a second job to handle the restart data to ultimately restart

```
ConfigureUpdateJob(RestartWanderJob.Schedule)
               .ScheduleOn(RestartData, BatchStrategy.MaximizeChunk)
               .RequireDataForIterateAsync(RestartData)
               .RequireDataForAddAsync(System.WanderInstanceData)
               .RequireDataForAddAsync(WanderStartData);
```

It works, but it's annoying and tedious. 

There's no reason the first job schedule where we read and write couldn't be done since they are different underlying data structures, they just share the same AccessController right now as we've grouped them together into a `VirtualData`. 

The question is "Should we split out the Cancelled aspects into it's own separate data structure".

We'd be introducing another job to handle consolidating that structure but we'd be saving on the ease of use when we want to use Cancelled Instances as a trigger to spawn new instances of the same type. 

This might dove tail into the below comment about consolidation in general.

Reply:
Hmmm, given what we discussed on the call with the ability to progress through a graceful cancel before informing your parent that you have cancelled I'm wondering if we treat cancellation as a form of failed resolution in the same way we resolve as failed when a task can't be completed (Ex: no path exists).

We'd just need to come up with a way to decern between failure to complete because impossible vs failure to complete because of cancel. If the parent could already be aware (some flag set) that it is within a cancellation flow then we don't need to write a reason for task failure (cancel vs impossible) into the result.


If you don't think what we discussed changes the nature of this problem then are we able to wrap types? Instead of a two phase job what if you had a generic struct that could hold the task data? 
Ex: `Cancelled<System.WanderInstanceData>`
That way you are dealing with different buckets without competing on the access controller?

Reply:
I reworked this so that we can read from cancelled data and write to the live data of the same type in the same job. No need to have this random `RestartData` struct to ping off of anymore.